src/java.base/windows/classes/sun/nio/ch/WindowsSelectorImpl.java

```diff
@@ -139,12 +140,11 @@ private MapEntry remove(SelectionKeyImpl ski) {
     WindowsSelectorImpl(SelectorProvider sp) throws IOException {
         super(sp);
         pollWrapper = new PollArrayWrapper(INIT_CAP);
-        wakeupPipe = Pipe.open();
+        wakeupPipe = new PipeImpl(sp, false);
         wakeupSourceFd = ((SelChImpl)wakeupPipe.source()).getFDVal();
 
         // Disable the Nagle algorithm so that the wakeup is more immediate
         SinkChannelImpl sink = (SinkChannelImpl)wakeupPipe.sink();
-        (sink.sc).socket().setTcpNoDelay(true);
         wakeupSinkFd = ((SelChImpl)sink).getFDVal();
```

Comment:
The "Disable the ..." comment can be removed as it's handled in PipeImpl now. "sink" can be removed so that wakeupSourceFd and wakeupSinkFd can be created the same way.

Reply:
Thanks, I pushed that change yesterday.